src/lib/utils/staticPaths.ts

```diff
@@ -1,21 +1,73 @@
-import { GetStaticPathsContext } from 'next'
 import { drupalClient } from '@/lib/utils/drupalClient'
 import { getAllPagedListingPaths } from '@/lib/utils/listingPages'
 import RESOURCE_TYPES from '@/lib/constants/resourceTypes'
+import { JsonApiResponse, JsonApiResourceWithPath } from 'next-drupal'
+
+async function getAllResourcesByResourceType(resourceType: string) {
+  const pageSize = 50 //must be <= 50 due to JSON:API limit
+  const params = {
+    // Note: we only need `path`, but we can't put it first:
+    // See:
+    //  https://next-drupal.org/guides/page-limit
+    //  https://dsva.slack.com/archives/C01SR56755H/p1695244241079879?thread_ts=1695070010.697129&cid=C01SR56755H
+    [`fields[${resourceType}]`]: 'title,path',
+    'page[limit]': pageSize,
+  }
+
+  // 1. Fetch first page.
+  const firstPage = await drupalClient.getResourceCollection<JsonApiResponse>(
+    resourceType,
+    {
+      params,
+      deserialize: false,
+    }
+  )
+
+  // 2. Determine number of pages to fetch.
+  const totalResourceCount = firstPage.meta.count
+  const firstPageData = drupalClient.deserialize(
+    firstPage
+  ) as JsonApiResourceWithPath[]
+  const pageCount = Math.ceil(totalResourceCount / pageSize)
+
+  // 3. If more pages, fetch them in parallel.
```

Comment:
This is my only worry with parallel fetches: content types that have a ton of pages will absolutely slam with requests here.

News Story: 2849 nodes
Events: 6936 nodes
VAMC Detail Page: 6384 nodes
VAMC Facility Health Care Service: 15099 nodes (not sure if these actually have standalone pages, though)


I guess the tradeoff is fetching in series is likely to be a fair bit slower... but we may want to look at some form of rate limiting here to prevent issues on these bulkier resource collections